superset-frontend/src/explore/components/controls/DatasourceControl/index.jsx

```diff
@@ -224,7 +263,7 @@ class DatasourceControl extends React.PureComponent {
       showSaveDatasetModal,
     } = this.state;
     const { datasource, onChange, theme } = this.props;
-    const isMissingDatasource = datasource.id == null;
+    const isMissingDatasource = datasource?.id == null;
```

Comment:
if datasource is undefined, is the expectation that this value should be true or false?

Reply:
`const isMissinDatasource = !!datasource.id` to make this truthy